stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/resiliency/loadbalance-connector.bal

```diff
@@ -0,0 +1,202 @@
+                               string httpVerb, LoadBalanceInferredConfig loadBalanceInferredConfig)
+(http:InResponse, http:HttpConnectorError) {
+    int httpVerbInt = extractHtpVerbIntValues(httpVerb);
+    return performLoadBalanceAction(path, outRequest, inRequest, httpVerbInt, loadBalanceInferredConfig);
+}
+
+// handles all the actions exposed through the Load Balance connector.
+function performLoadBalanceAction (string path, http:OutRequest outRequest, http:InRequest inRequest,
+                                int requestAction, LoadBalanceInferredConfig loadBalanceInferredConfig)
+(http:InResponse, http:HttpConnectorError) {
+
+    endpoint<http:HttpClient> endPoint {
+
+    }
+
+    http:HttpClient loadBlanceClient = loadBalanceInferredConfig.algorithm(loadBalanceInferredConfig.loadBalanceClientsArray);
+
+    bind loadBlanceClient with endPoint;
+    LoadBalanceConnectorError loadBalanceConnectorError = {};
+    http:HttpConnectorError httpConnectorError;
+    loadBalanceConnectorError.httpConnectorError = [];
+    http:InResponse inResponse;
+
+    // When performing passthrough scenarios using Load Balance connector, message needs to be built before trying out the
+    // load balance endpoints to keep the request message to load balance the messages in case of failure.
+    if (inRequest != null && FORWARD_INT == requestAction) {
+        blob binaryPayload = inRequest.getBinaryPayload();
+    }
+
+    inResponse, httpConnectorError = invokeEndpoint(path, outRequest, inRequest, requestAction, loadBlanceClient);
+
+    while (httpConnectorError != null) {
+        loadBalanceConnectorError.httpConnectorError[nextIndex] = httpConnectorError;
+        loadBlanceClient = loadBalanceInferredConfig.algorithm(loadBalanceInferredConfig.loadBalanceClientsArray);
+        bind loadBlanceClient with endPoint;
+        inResponse, httpConnectorError = invokeEndpoint(path, outRequest, inRequest, requestAction, loadBlanceClient);
+        terminatenumber = terminatenumber + 1;
+        if (terminatenumber == (lengthof loadBalanceInferredConfig.loadBalanceClientsArray - 1) && inResponse == null) {
+            terminatenumber = 0;
+            inResponse, httpConnectorError = populateGenericLoadBalanceConnectorError(loadBalanceConnectorError, httpConnectorError, nextIndex);
+            break;
+        }
+    }
+    terminatenumber = 0;
+    return inResponse, httpConnectorError;
+}
+
+// Round Robin Algothim implementation with respect to load balancing endpoints.
+public function (http:HttpClient [])(http:HttpClient) roundRobin =
+                        function (http:HttpClient [] loadBalaceConfigArray) returns (http:HttpClient nextEndpoint) {
+
+     http:HttpClient httpClient;
+
+     if (nextIndex == ((lengthof (loadBalaceConfigArray)) - 1)) {
+         httpClient = loadBalaceConfigArray[nextIndex];
+         nextIndex = 0;
+     } else {
+         httpClient = loadBalaceConfigArray[nextIndex];
+         nextIndex = nextIndex + 1;
+     }
```

Comment:
We can simplify this by taking the modulus of the index right?

Reply:
This cannot be simplified using modulus due index values will not reset correctly.